pkg/util/annotations/annotations.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	"strings"
+)
+
+const (
+	singleAnnotationPrefix = "metadata.annotations['"
+	singleAnnotationSuffix = "']"
+)
+
+// GetSingleAnnotationKey returns the key in the specified single annotation
+// field path, or an error if the fieldPath is not a valid single annotation or
+// the key extracted from the fieldPath is not valid.
+//
+// Example inputs and outputs:
+//  - "metadata.annotations['myKey']" --> ("myKey", true)
+//  - "metadata.labels --> ("", false)
+//  - "metadata.annotations --> ("", false)
+//  - "metadata.annotations['a[b]c'] --> ("", false)
+func GetSingleAnnotationKey(fieldPath string) (string, bool) {
```

Comment:
Sorry to beat this one over and over, but it's still awkward.  This whole package feels wrong.

Why is this not in the fieldpath package, something like:

fieldpath.SplitMaybeSubscriptedPath(value string) (path string, subscript string, isSubscript bool)

?

Reply:
Just to make sure I understand this, do the following input/output reflect what this function does?

`"metadata.annotations" --> ("metadata.annotations", "", false)`
`"metadata.annotations['myKey']" --> ("metadata.annotations", "myKey", true)`
`"metadata.annotations['a[b]c']" --> ("metadata.annotations['a[b]c']", "", false)`

Reply:
And, yeah, `fieldpath` package is a much better place.

Reply:
I thin the third case can be argued, but the simpler model would produce `("metadata.annotations"'a[b]c'", false)` and let the caller validate that the field is annotations and the key is a valid value.

Reply:
Done. Please take a look at this new version.

I now don't think we should require any characters (`'`, `[` and `]`) in the key to be escaped, so I remove the check and only verify the key can pass `validate.IsQualifiedName()`.